### ISSUE TITLE
structcheck returns unused false positives for fields of an aliased type

### ISSUE THREAD

**Issue by andrewpmartinez** [golangci/golangci-lint#572]
The following code is an example that is triggering unused errors from structcheck
```golang
type unaryOp struct {
	Identifier string
	value      interface{}
}

type Contains unaryOp

func newContains(identifier, value string) Contains {
	return Contains{
		Identifier: identifier,
		value:      value,
	}
}

func (c Contains) ToSql() (string, []interface{}, error) {
	v := fmt.Sprintf("%%%s%%", c.value)

	args := []interface{}{v}

	sql := fmt.Sprintf("%s NOT LIKE %s", c.Identifier, "?")

	return sql, args, nil
}
```

Please include the following information:
1. Version of golangci-lint: `golangci-lint --version` (or git commit if you don't use binary distribution)

```3d954ea19dd4e16e1dd987e3a1715c9a7b9b135a```
head of master on June 12th 2019

2. Config file: `cat .golangci.yml`

Not in use

3. Go environment: `go version && go env`
```go version go1.12 windows/amd64
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\andre\AppData\Local\go-build
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOOS=windows
set GOPATH=D:\Source\go
set GOPROXY=
set GORACE=
set GOROOT=C:\Go
set GOTMPDIR=
set GOTOOLDIR=C:\Go\pkg\tool\windows_amd64
set GCCGO=gccgo
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\andre\AppData\Local\Temp\go-build083896558=/tmp/go-build -gno-record-gcc-switches```


4. Verbose output of running: `golangci-lint run -v`
```level=info msg="[config_reader] Config search paths: [./ D:\\Source\\go\\src\\redacted\\redacted\\redacted\\redacted D:\\Source\\go\\src\\redacted\\redacted\\redacted D:\\Source\\go\\src\\redacted\\redacted D:\\Source\\go\\src\\redacted D:\\Source\\go\\src D:\\Source\\go D:\\Source D:\\]"
level=info msg="[lintersdb] Active 10 linters: [deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck]"
level=info msg="[lintersdb] Optimized sublinters [staticcheck gosimple unused] into metalinter megacheck"
level=info msg="[loader] Go packages loading at mode load types and syntax took 1.9789972s"
level=info msg="[loader] SSA repr building timing: packages building 18.9997ms, total 208.0021ms"
level=info msg="[runner] worker.6 took 0s with stages: typecheck: 0s"
level=info msg="[runner] worker.2 took 12.9998ms with stages: structcheck: 12.9998ms"
level=info msg="[runner] worker redacted\redacted\predicate\ops.go:20:2: `value` is unused (structcheck)
        value      interface{}
        ^
r.1 took 15.0001ms with stages: errcheck: 15.0001ms"
level=info msg="[runner] worker.5 took 46.9989ms with stages: varcheck: 46.9989ms"
level=info msg="[runner] worker.3 took 136.9996ms with stages: deadcode: 136.9996ms"
level=info msg="[runner] worker.8 took 581.0004ms with stages: ineffassign: 581.0004ms"
level=info msg="[runner] worker.7 took 590.9992ms with stages: govet: 590.9992ms"
level=info msg="[runner] worker.4 took 1.953s with stages: megacheck: 1.953s"
level=info msg="[runner] Workers idle times: #1: 1.802002s, #2: 1.9380014s, #3: 1.7810013s, #5: 1.7910024s, #6: 1.9490002s, #7: 1.3490027s, #8: 1.3720024s"
level=info msg="[runner] Issues before processing: 127, after processing: 1"
level=info msg="[runner] processing took 3.9977ms with stages: skip_dirs: 2.9972ms, identifier_marker: 1.0005ms, exclude: 0s, diff: 0s, cgo: 0s, filename_unadjuster: 0s, path_prettifier: 0s, skip_files: 0s, uniq_by_line: 0s, max_per_file_from_linter: 0s, autogenerated_exclude: 0s, max_same_issues: 0s, source_code: 0s, path_shortener: 0s, exclude-rules: 0s, nolint: 0s, max_from_linter: 0s"
level=info msg="File cache stats: 1 entries of total size 2.0KiB"
level=info msg="Memory: 45 samples, avg is 673.4MB, max is 1632.7MB"
level=info msg="Execution took 4.6860006s"```

**Comment by Helcaraxan** [comment]
A different shape of this same issue occurs when embedded a `struct` in another one. If a field is only ever referenced via the outer embedding `struct` a false positive will be raised.
